Documentation/contributor-guide/features.md

```diff
@@ -0,0 +1,60 @@
+# Features 
+
+This document provides an overview of etcd features and general guidelines for adding and removing features. The project maintainers can override the guidelines per the need of the project following the project governance.
+
+The etcd features fall into three stages, experimental, stable, and unsafe.
+
+## Experimental
+
+Any new feature should be added as an experimental feature. An experimental feature is characterized as below:
+- Might be buggy due to lack of user testing. Enabling the feature may not work as expected.
+- Support for such a feature may be dropped at any time without notice
```

Comment:
I think we should clarify what dropping support means. This is important to convey what users should expect when we break support. I think there are two parts for it:
* Issues related to the feature will be closed without investigation
* Feature will be removed in the next minor release.